lib/EasyDirectiveVisitor.ts

```diff
@@ -359,8 +359,17 @@ abstract class EasyDirectiveVisitor<
     query: GraphQLObjectType<any, TContext>,
     schema: GraphQLSchema,
     directiveName: string,
-  ): TLocation extends DirectiveLocation.QUERY ? void : never {
-    throw new Error('Method not implemented.');
+  ): GraphQLObjectType<any, TContext> {
+    return query;
+  }
+
+  // istanbul ignore next (should be overridden and never reached)
+  public visitMutation(
+    mutation: GraphQLObjectType<any, TContext>,
+    schema: GraphQLSchema,
+    directiveName: string,
+  ): GraphQLObjectType<any, TContext> {
+    return mutation;
```

Comment:
nit: Wouldn't it be better to have these methods throw an error if not overwritten by the directives using that can be defined in these locations ?

Reply:
@DDDKnightmare the problem here is that `visitQuery` and `visitMutation` are always called, they search for arguments and input objects, marking them to be validated before wrapping field resolvers. So directives that does not need to do this search would have to implement these methods anyway only returning the query/mutation untouched.

https://github.com/profusion/apollo-validation-directives/blob/906e8317b0dd5f4770ac158e9e1aa0e5c6d19174/lib/createSchemaMapperForVisitor.ts#L20-L27